controllers/mirrorpeer_controller.go

```diff
@@ -59,6 +61,27 @@ const hubRecoveryLabel = "cluster.open-cluster-management.io/backup"
 //+kubebuilder:rbac:groups=addon.open-cluster-management.io,resources=managedclusteraddons;clustermanagementaddons,verbs=*
 //+kubebuilder:rbac:groups=addon.open-cluster-management.io,resources=managedclusteraddons/status,verbs=*
 
+// Checks whether a string is contained within a slice
+func contains(slice []string, s string) bool {
+	for _, item := range slice {
+		if item == s {
+			return true
+		}
+	}
+	return false
+}
+
+// Removes a given string from a slice and returns the new slice
```

Comment:
Move this too.
Also check if there is any native utility package we could use for this